src/utils/customPropTypes.js

```diff
@@ -0,0 +1,16 @@
+import _ from 'lodash';
+
+let customPropTypes = {
+  mutuallyExclusive: (exclusives, props, propName, componentName) => {
+    _.forEach(exclusives, exclusiveProp => {
+      if (props[propName] && props[exclusiveProp]) {
+        throw new Error(`\`${componentName}\` should only have one of type \`${propName}\` or \`${exclusiveProp}\`, ` +
+          `not both.`);
+      }
```

Comment:
In order to compare the specific propType to the exclusive props passed in, I had to take advantage React's custom validation function found in [this article](https://facebook.github.io/react/docs/reusable-components.html) - so we have to bind the `exclusives` to this function in order to access and take advantage of the additional params `props` `propName` and `componentName`. 

This way, we have the current prop to compare the `exclusives` to.